tests/impl.cpp

```diff
@@ -7137,7 +7137,27 @@ result_t test_mm_maddubs_epi16(const SSE2NEONTestImpl &impl, uint32_t iter)
 
 result_t test_mm_maddubs_pi16(const SSE2NEONTestImpl &impl, uint32_t iter)
 {
-    return TEST_UNIMPL;
+    const uint8_t *_a = (const uint8_t *) impl.mTestIntPointer1;
+    const int8_t *_b = (const int8_t *) impl.mTestIntPointer2;
+    int16_t d0 = (int16_t)(_a[0] * _b[0]);
+    int16_t d1 = (int16_t)(_a[1] * _b[1]);
+    int16_t d2 = (int16_t)(_a[2] * _b[2]);
+    int16_t d3 = (int16_t)(_a[3] * _b[3]);
+    int16_t d4 = (int16_t)(_a[4] * _b[4]);
+    int16_t d5 = (int16_t)(_a[5] * _b[5]);
+    int16_t d6 = (int16_t)(_a[6] * _b[6]);
+    int16_t d7 = (int16_t)(_a[7] * _b[7]);
+
+    int16_t e0 = saturate_16(d0 + d1);
+    int16_t e1 = saturate_16(d2 + d3);
+    int16_t e2 = saturate_16(d4 + d5);
+    int16_t e3 = saturate_16(d6 + d7);
```

Comment:
`saturate_s16()` should have argument of `int32_t`.
`int16_t` + `int16_t` would produce `int16_t` only.

Reply:
The result of `d0 + d1` would surpass the range of `int16_t`, since argument `a` is treated as `uint8_t` and `b` is treated as `int8_t`. The maximum positive value of `int16_t` is `2^15 - 1`, and the maximum possible result would be `2*(2^8 - 1)*(2^7 - 1)` which would be more than `2^15 - 1`. Therefore, we need `saturate_16()`

Reply:
I might misread in the previous review.